hr_timesheet_sheet_no_negative/models/hr_timesheet_sheet.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2019 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models, _
+
+
+class SheetLine(models.TransientModel):
+    _inherit = 'hr_timesheet.sheet.line'
+
+    def _negative_unit_amount(self):
+        if not self.sheet_id.company_id:
+            if self._context.get('company_id'):
+                company_id = self._context.get('company_id')
+                company = self.env['res.company'].browse(company_id)
+            else:
+                company = self.sheet_id.employee_id.company_id
+        else:
+            company = self.sheet_id.company_id
+        return company.timesheet_negative_unit_amount
+
+    @api.onchange('unit_amount')
+    def onchange_unit_amount(self):
+        self.ensure_one()
+        sheet = self._get_sheet()
+        if not sheet:
+            return {'warning': {
+                'title': _("Warning"),
+                'message': _("Save the Timesheet Sheet first.")
+            }}
+        if not self._negative_unit_amount() and self.unit_amount < 0.0:
+            self.write({'unit_amount': 0.0})
+        super(SheetLine, self).onchange_unit_amount()
```

Comment:
Since you are inheriting from an onchange, here you should return the result of the super, otherwise you could break some eventual other `onchange_unit_amount()` customization.